src/analysis/retail/monk/mistweaver/modules/spells/JadeEmpowerment.tsx

```diff
@@ -0,0 +1,108 @@
+import SPELLS from 'common/SPELLS';
+import { TALENTS_MONK } from 'common/TALENTS';
+import { SpellLink } from 'interface';
+import { explanationAndDataSubsection } from 'interface/guide/components/ExplanationRow';
+import { RoundedPanel } from 'interface/guide/components/GuideDivs';
+import Analyzer, { Options, SELECTED_PLAYER } from 'parser/core/Analyzer';
+import Events, { ApplyBuffEvent, RefreshBuffEvent } from 'parser/core/Events';
+import { GUIDE_CORE_EXPLANATION_PERCENT } from '../../Guide';
+import JadefireTeachings from './JadefireTeachings';
+import { BoxRowEntry, PerformanceBoxRow } from 'interface/guide/components/PerformanceBoxRow';
+import { QualitativePerformance } from 'parser/ui/QualitativePerformance';
+
+class JadeEmpowerment extends Analyzer {
+  protected jft!: JadefireTeachings;
+  wastedCharges: number = 0;
+  castEntries: BoxRowEntry[] = [];
+
+  constructor(options: Options) {
+    super(options);
+    this.active = this.selectedCombatant.hasTalent(TALENTS_MONK.CHI_BURST_SHARED_TALENT);
+    this.addEventListener(
+      Events.refreshbuff.by(SELECTED_PLAYER).spell(SPELLS.JADE_EMPOWERMENT_BUFF),
+      this.onRefresh,
+    );
+    this.addEventListener(
+      Events.applybuff.by(SELECTED_PLAYER).spell(SPELLS.JADE_EMPOWERMENT_BUFF),
+      this.onApply,
+    );
+  }
+
+  onApply(event: ApplyBuffEvent) {
```

Comment:
I think it makes more sense to hook these off of CJL casts and also check that JFT is active when casting to qualify the buff efficiency as Good or Bad.
Keeping the refresh listener is good to mark those as bad